lib/net/nforceif/include/lwipopts.h

```diff
@@ -43,7 +43,9 @@
 #define LWIP_IPV4 1
 #define LWIP_IPV6 1
 #define LWIP_DEBUG 1
-extern int errno;
+#define LWIP_ERRNO_INCLUDE <errno.h>
+#define LWIP_COMPAT_MUTEX_ALLOWED
```

Comment:
http://git.savannah.nongnu.org/cgit/lwip.git/commit/?id=42dfa71f978296edd79040641234d9496f1439a7 introduced a change that makes the build error out when the platform does not provide glue code for using mutexes. This should definitely get resolved by adding such code in the future, but for now, this keeps the current behavior.
Also see the comment below.